ttnn/cpp/ttnn/tensor/tensor_layout.hpp

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+#include "enum_types.hpp"
+
+#include <cstddef>
+#include <ostream>
+
+namespace tt::tt_metal {
+
+class Size {
+public:
+    Size(size_t height, size_t width);
+    Size(const std::pair<size_t, size_t>& size);
+    Size(const std::array<size_t, 2>& size);
+
+    operator std::pair<size_t, size_t>() const;
+    operator std::array<size_t, 2>() const;
+    operator std::array<uint32_t, 2>() const;
+
+    Size operator/(const Size& rhs) const;
+    Size operator*(size_t scalar) const;
+
+    // comparison operator
+    bool operator==(const Size& rhs) const;
+
+    size_t height() const;
+    size_t width() const;
+
+    // does not have to be a member, but it is easier to find if it is
+    Size aligned_to_tile(const Size& tile);
+
+private:
+    size_t mHeight = 0;
+    size_t mWidth = 0;
+};
+
+std::ostream& operator<<(std::ostream& os, const tt::tt_metal::Size& size);
+
+// Creating as a class to differentiate the use between LegacyPadding represented as SimpleShape and Alignment.
+// This class has to eventually become its own class
+class Alignment : public ttnn::SimpleShape {
```

Comment:
I would prefer to not publicly inherit Alignment from SimpleShape, because this way Alignment may be implicitly casted to SimpleShape. Moreover, SimpleShape doesn't have a virtual destructor, so if someone will accidentally add another member to Alignment class, thing may just fall apart

Reply:
100% agree, was a temporary quick hack, not going to be merged this way